integration-test/1491-brazil-shields.py

```diff
@@ -0,0 +1,827 @@
+                'incorrect:name': 'MGC-259', 'oneway': 'no',
+                'ref': 'MGC-259;BR-259', 'highway': 'primary',
+            }),
+            dsl.relation(1, {
+                'type': 'route', 'route': 'road', 'ref': 'BR-259',
+                'network': 'BR', 'source': 'openstreetmap.org',
+            }, ways=[31574746]),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 31574746,
+                'all_shield_texts': ['259', '259'],
+                'all_networks': ['BR', 'BR:MG:connecting'],
+            })
+
+    def test_br_CMG(self):
+        import dsl
+
+        z, x, y = (16, 24787, 35853)
+
+        self.generate_fixtures(
+            dsl.is_in('BR', z, x, y),
+            # https://www.openstreetmap.org/way/31574975
+            dsl.way(31574975, dsl.tile_diagonal(z, x, y), {
+                'name': u'Avenida Governador Magalh\xe3es Pinto',
+                'IBGE:CD_ADMINIS': 'federal', 'surface': 'paved',
+                'source': 'openstreetmap.org', 'oneway': 'yes',
+                'ref': 'CMG-251;BR-251', 'highway': 'primary',
+            }),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 31574975,
+                'all_shield_texts': ['251', '251'],
+                'all_networks': ['BR', 'BR:MG'],
+            })
+
+    def test_br_ERS(self):
+        import dsl
+
+        z, x, y = (16, 23101, 38967)
+
+        self.generate_fixtures(
+            dsl.is_in('BR', z, x, y),
+            # https://www.openstreetmap.org/way/20906150
+            dsl.way(20906150, dsl.tile_diagonal(z, x, y), {
+                'old_ref': 'RS-602', 'maxspeed': '80', 'surface': 'asphalt',
+                'source': 'openstreetmap.org', 'ref': 'ERS-602',
+                'highway': 'secondary',
+            }),
+            dsl.relation(1, {
+                'network': 'BR:RS', 'ref': 'ERS-602', 'route': 'road',
+                'surface': 'asphalt', 'source': 'openstreetmap.org',
+                'type': 'route',
+            }, ways=[20906150]),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 20906150,
+                'shield_text': '602', 'network': 'BR:RS:state',
+                'all_networks': ['BR:RS:state'],
+                'all_shield_texts': ['602'],
+            })
+
+    def test_br_VRS(self):
+        import dsl
+
+        z, x, y = (16, 23312, 38227)
+
+        self.generate_fixtures(
+            dsl.is_in('BR', z, x, y),
+            # https://www.openstreetmap.org/way/26757190
+            dsl.way(26757190, dsl.tile_diagonal(z, x, y), {
+                'source': 'openstreetmap.org', 'ref': 'VRS-851',
+                'highway': 'secondary', 'oneway': 'yes',
+            }),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 26757190, 'shield_text': '851',
+                'network': 'BR:RS:secondary',
```

Comment:
What's your reference for saying `BR:RS:secondary` versus just `BR:RS`? 

I do see it listed as a "secondary highway" on https://en.wikipedia.org/wiki/List_of_state_highways_in_Rio_Grande_do_Sul.

But it's not supported by other map providers &/or photos and I think "secondary" here still means state route – since we're not drilling down on any of the other "subtypes" of RS roads so I think the network should be `BR:RS`?

Reply:
I added it on the basis that `VRS` was giving us more information than just `RS`, and we could always ignore it by matching on `BR:RS.*`.

There's a [list of roads in Rio Grande do Sul](https://pt.wikipedia.org/wiki/Lista_de_rodovias_do_Rio_Grande_do_Sul#Rodovias_vicinais) on the Portuguese Wikipedia, which lists the `VRS` as "Rodovias vicinais", which seems to translate to something like "secondary", "local" or "service".

I've removed it in c68d6eb, but easy to add back again if we want.